backends/graphics/openglsdl/openglsdl-graphics.cpp

```diff
@@ -352,37 +348,54 @@ bool OpenGLSdlGraphicsManager::loadVideoMode(uint requestedWidth, uint requested
 		// Set the window size to the values stored when the window was maximized
-		// for the last time. We also need to reset any scaling here.
+		// for the last time.
 		requestedWidth  = ConfMan.getInt("window_maximized_width", Common::ConfigManager::kApplicationDomain);
 		requestedHeight = ConfMan.getInt("window_maximized_height", Common::ConfigManager::kApplicationDomain);
 
-	} else if (ConfMan.hasKey("last_window_width", Common::ConfigManager::kApplicationDomain) && ConfMan.hasKey("last_window_height", Common::ConfigManager::kApplicationDomain)) {
-		// Restore previously stored window dimensions.
+	} else if (!_isMaximized && ConfMan.hasKey("last_window_width", Common::ConfigManager::kApplicationDomain) && ConfMan.hasKey("last_window_height", Common::ConfigManager::kApplicationDomain)) {
+		// Load previously stored window dimensions.
 		requestedWidth  = ConfMan.getInt("last_window_width", Common::ConfigManager::kApplicationDomain);
 		requestedHeight = ConfMan.getInt("last_window_height", Common::ConfigManager::kApplicationDomain);
 
 	} else {
 		// Set the basic window size based on the desktop resolution
 		// since we have no values stored, e.g. on first launch.
-		requestedWidth  = desktopRes.width()  * 0.3f;
-		requestedHeight = desktopRes.height() * 0.4f;
-
-		// Apply scaler
-		requestedWidth  *= _graphicsScale;
-		requestedHeight *= _graphicsScale;
+		requestedWidth  = MAX<uint>(desktopRes.width() / 2, MIN<uint>(640, desktopRes.width()));
+		requestedHeight = MIN<uint>(requestedWidth * 3 / 4, desktopRes.height());
 
 		// Save current window dimensions
 		ConfMan.setInt("last_window_width", requestedWidth, Common::ConfigManager::kApplicationDomain);
 		ConfMan.setInt("last_window_height", requestedHeight, Common::ConfigManager::kApplicationDomain);
 		ConfMan.flushToDisk();
 	}
 
-	// Determine current aspect ratio
+#else
+		// Set the basic window size based on the desktop resolution
+		// since we cannot reliably determine the current window state
+		// on SDL1.
+		requestedWidth  = MAX<uint>(desktopRes.width() / 2, MIN<uint>(640, desktopRes.width()));
+		requestedHeight = MIN<uint>(requestedWidth * 3 / 4, desktopRes.height());
+
+#endif
+
+	// In case we detected an increase in the requested window dimensions (e.g. when launching
+	// a game in 800x600 while having a smaller screen size stored in the configuration file),
+	// we override the window dimensions with the "real" request.
+	if ((requestedWidth < _lastRequestedWidth || requestedHeight < _lastRequestedHeight) && ConfMan.getActiveDomain()) {
+		requestedWidth  = _lastRequestedWidth  * _graphicsScale;
+		requestedHeight = _lastRequestedHeight * _graphicsScale;
+	}
+
+	// Set allowed dimensions
 	uint maxAllowedWidth   = desktopRes.width();
 	uint maxAllowedHeight  = desktopRes.height();
-	float ratio = (float)requestedWidth / requestedHeight;
```

Comment:
See comment above. I think we probably want to use `float ratio = (float) _lastRequestedWidth / (float) _lastRequestedHeight;` here rather than the ratio computed above from the desktop resolution.

Reply:
Ha, I think I understand now why it was using `requestedWidth / requestedHeight` and not `_lastRequestedWidth / _lastRequestedHeight`. With the later you could end up with a size that is still bigger than the allowed size if decreasing the height causes the width to increase due to the requested ratio being wider than the computed one.

So reinstating the line above seems actually the better option.

Reply:
Thanks, I moved the aspect ratio calculation back to it's old place.